src/inbox/mod_inbox_muc.erl

```diff
@@ -0,0 +1,94 @@
+%%%-------------------------------------------------------------------
+%%% @author Dominik Stanaszek dominik.stanaszek@erlang-solutions.com
+%%% @copyright (C) 2018, Erlang-Solutions
+%%% @doc
+%%%
+%%% @end
+%%% Created : 6.07.2018 
+%%%-------------------------------------------------------------------
+-module(mod_inbox_muc).
+-author("dominik.stanaszek@erlang-solutions.com").
+-include("jlib.hrl").
+
+-export([update_inbox/5, start/1, stop/1]).
+
+start(Host) ->
+    ejabberd_hooks:add(update_inbox, Host, ?MODULE, update_inbox, 90),
+    % TODO check ooptions: if system messages stored -> 
+    % add hook handler for system messages on hook ie. invitation_sent
+    ok.
+
+stop(Host) ->
+    ejabberd_hooks:delete(update_inbox, Host, ?MODULE, update_inbox, 90),
+    ok.
+
+update_inbox(Acc, Room, {From, FromRoomJid}, AffsDict, Packet) ->
+    Affs = dict:to_list(AffsDict),
+    Users = affs_to_allowed_users(Affs),
+    FromBare = jid:to_bare(From),
+    lists:foreach(fun(User) -> update_inbox(Room, 
```

Comment:
Can we use list comprehension instead? It's not big deal but It's just more readable:
```
[update_inbox(room, {FromBare, FromRoomJid}, User, Packet) || User <- Users]
```

Reply:
I will refactor this next sprint, I agree it's better.